src/main/java/org/jfrog/bamboo/util/TaskDefinitionHelper.java

```diff
@@ -82,9 +81,13 @@ public static TaskDefinition findMavenDefinition(List<? extends TaskDefinition>
     @Nullable
     public static TaskDefinition findReleaseTaskDefinition(List<? extends TaskDefinition> taskDefinitions) {
         if (taskDefinitions != null) {
-            for (TaskDefinition definition : taskDefinitions) {
-                if ("true".equals(definition.getConfiguration().get(ENABLE_RELEASE_MANAGEMENT))) {
-                    return definition;
+            for (TaskDefinition taskDefinition : taskDefinitions) {
+                if (taskDefinition.isEnabled()) {
+                    AbstractBuildContext config = AbstractBuildContext.createContextFromMap(taskDefinition.getConfiguration());
+                    // Check the release management is enabled
+                    if ((config != null) && config.releaseManagementContext.isReleaseMgmtEnabled()) {
```

Comment:
Redundant parentheses around ```(config != null)```.
Also, this method is also identical to the ```getTaskConfiguration``` method. I suggest reusing the code.

Reply:
Per our discussion, the config can be null so this check is not redundant. I did a reuse for the code.